CMakeLists.txt

```diff
@@ -164,30 +164,50 @@ find_package(CURL REQUIRED)
 include_directories(${CURL_INCLUDE_DIRS})
 link_directories(${CURL_LIBRARY_DIRS})
 
-# set(Python_ADDITIONAL_VERSIONS 3.6 3.5 3.4)
-
-find_package(PythonInterp)
-find_package(PythonLibs)
+if (${CMAKE_VERSION} VERSION_LESS "3.13.3")
+    find_package(PythonInterp)
+    set(Python_Interpreter_FOUND ${PYTHONINTERP_FOUND})
+    set(Python_EXECUTABLE ${Python_EXECUTABLE})
+    set(Python_VERSION ${PYTHON_VERSION_STRING})
+    set(Python_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
+    set(Python_LIBRARY_DIRS ${PYTHON_LIBRARY_DIRS} ${NUMPY_LIBRARY_DIRS})
+    set(Python_VERSION_MAJOR ${PYTHON_VERSION_MAJOR})
+    find_package(PythonLibs)
+    set(Python_Development_FOUND ${PYTHONLIBS_FOUND})
+    if (PYTHONLIBS_FOUND)
+        if (${PYTHONLIBS_VERSION_STRING} VERSION_LESS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}.0")
+            message(WARNING "Python library version found does not match python interpreter version. Specify PYTHON_LIBRARY and PYTHON_INCLUDE_DIR manually.")
+        endif()
+        set(Python_LIBRARIES ${PYTHON_LIBRARIES})
+        find_package(NumPy)
+        if(NUMPY_FOUND)
+            set(Python_NumPy_FOUND TRUE)
+            set(Python_NumPy_INCLUDE_DIRS ${NUMPY_INCLUDE_DIRS})
+            set(Python_NumPy_VERSION ${NUMPY_VERSION})
+            set(Python_FOUND TRUE)
+        endif(NUMPY_FOUND)
+    else()
+        unset(PYTHON_LIBRARIES)
+    endif (PYTHONLIBS_FOUND)
+    set(Python_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
+    set(Python_LIBRARY_DIRS ${PYTHON_LIBRARY_DIRS})
+    set(Python_NumPy_INCLUDE_DIRS ${NUMPY_INCLUDE_DIRS})
+    set(Python_VERSION_MAJOR ${PYTHON_VERSION_MAJOR})
+else()
+    find_package(Python COMPONENTS Interpreter Development NumPy)
+endif()
 
 if (PYTHON_PLUGIN)
-    if (NOT ${PYTHON_VERISON_MAJOR} EQUAL ${OpenCV_VERSION_MAJOR})
-        message(WARNING "Python version " ${PYTHON_VERSION_MAJOR} " incompatible with OpenCV version " ${OpenCV_VERSION_MAJOR})
+    if (${Python_VERISON_MAJOR} LESS ${OpenCV_VERSION_MAJOR})
```

Comment:
Typo in Python_VERISON_MAJOR, should be Python_VERSION_MAJOR

Reply:
Thanks!